pkg/inst-api-semconv/instrumenter/http/http_attrs_extractor.go

```diff
@@ -0,0 +1,118 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package http
+
+import (
+	"context"
+	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/pkg/inst-api-semconv/instrumenter/utils"
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.30.0"
+	"sync/atomic"
+)
+
+type HttpRequest interface{}
+
+type HttpResponse interface{}
+
+type HttpCommonAttrsExtractor[REQUEST HttpRequest, RESPONSE HttpResponse, GETTER1 HttpCommonAttrsGetter[REQUEST, RESPONSE]] struct {
+	HttpGetter       GETTER1
+	AttributesFilter func(attrs []attribute.KeyValue) []attribute.KeyValue
+}
+
+func (h *HttpCommonAttrsExtractor[REQUEST, RESPONSE, GETTER1]) OnStart(attributes []attribute.KeyValue, parentContext context.Context, request REQUEST) ([]attribute.KeyValue, context.Context) {
+	attributes = append(attributes, attribute.KeyValue{
+		Key:   semconv.HTTPRequestMethodKey,
+		Value: attribute.StringValue(h.HttpGetter.GetRequestMethod(request)),
```

Comment:
What if h.HttpGetter.GetRequestMethod(request) throws exception or contains unexpected return value?

Reply:
Getter should ensure that it does not panic and return any unexpected value.